include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -66,7 +95,9 @@ class StatsdClient {
                  const std::string& prefix,
                  const uint64_t batchsize = 0,
                  const uint64_t sendInterval = 1000,
-                 const int gaugePrecision = 4) noexcept;
+                 const int gaugePrecision = 4,
+                 FrequencyFunc frequencyFunc =
+                     std::bind(detail::rand, std::random_device()())) noexcept;
```

Comment:
we no longer have a seed method on the class but rather you pass in a sampling function. i personally like the term sample better here but because it has interplay with the "frequency" (nomenclature from statsd proper) i chose to make them match